### PR TITLE
Text data extraction

### PR DESCRIPTION
**Pull request by yuvve**
Data extractor class implemented:
- Can extract data from either file or raw HTML
- Can extract with different amount of filtering
- Implemented tests for tag residue

### CODE REVIEW

File: extract.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+to reduce noise in AI model training.
+"""
+import logging
+import copy
+import regex as re
+from bs4 import BeautifulSoup
+
+class NoSoup(Exception):
+    """
+    Data extractor hasn't generated a soup, and cannot start extracting data.
```
Comment (Hedlund01) [inline extract.py:12]: Look over this message, why would the extractor want a soup haha?
Make it even more specific by saying that some BeautifulSoup stuff is missing and not soup.

File: tests/extract_tests.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    Requests a URL, then runs following tests:
+    - Check if tags exists in raw HTML; expected: True
+    - Check if no tags in unfiltered, extracted data; expected: False
+    - Check if no tags in filtered, extracted data; expected: False
+    - Check if no tags in filtered, paragraph only, extracted data; expected: False
+    """
+    r = requests.get(address)
+    extractor = DataExtractor()
+    extractor.create_soup_from_html(r.text)
```
Comment (Hedlund01) [inline tests/extract_tests.py:33]: Why aren't you testing with the open_file function instead, as that both tests the create_soup_from_html function and the open_file function. Now you have a function that isn't tested.

File: tests/extract_tests.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import regex as re
+from extract import DataExtractor
+
+def exist_tags_in_plaintext(text):
+    """
+    Check if the plaintext contains any HTML tags
+    :param text: plaintext
+    :returns: boolean
+    """
+    tag_re = re.compile(r'</?\s*[a-z-][^>]*\s*>|(\&(?:[\w\d]+|#\d+|#x[a-f\d]+);)')
```
Comment (Hedlund01) [inline tests/extract_tests.py:18]: Add comment about what the regex actually do

File: extract.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        Opens a file and creates a soup object from a file.
+        :param file_path: path to an html file.
+        """
+        try:
+            with open(file_path, encoding='utf-8') as fp:
+                self.create_soup_from_html(fp)
+        except (FileNotFoundError):
+            logging.error("Couldn't find %s!", file_path)
+    
+    def create_soup_from_html(self, raw_html):
```
Comment (Hedlund01) [inline extract.py:40]: This function should be private

**Review by Hedlund01 — CHANGES_REQUESTED** [review]

File: extract.py (hunk 4 above)
Comment (yuvve) [reply extract.py:40]: As discussed, we change the name of the function instead of marking it as private.

**Comment by yuvve** [comment]
Updated code according to @Hedlund01 comments.

**Review by F1sshy — APPROVED** [review]
Looks good

**Review by Hedlund01 — APPROVED** [review]
LGTM

**Pull request merged**